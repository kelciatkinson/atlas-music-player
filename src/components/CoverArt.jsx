import placeholder from "../assets/placeholder.svg";

export default function CoverArt() {
  return <div className="pb-5">
    <img src={placeholder} alt="placeholder" className="block max-w-100"></img>
  </div>
}
