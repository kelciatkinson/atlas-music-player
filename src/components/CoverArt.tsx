// import placeholder from "../assets/placeholder.svg";
// import music from "../assets/music.jpg";
import React, { useEffect, useState } from "react";


interface CoverArtProps {
  cover: string;
  songId?: string;
}

export default function CoverArt({ cover, songId }: CoverArtProps) {
  const [lyrics, setLyrics] = useState<string>("");
  const [showLyrics, setShowLyrics] = useState<boolean>(false);


  const fetchLyrcis = async () => {
    if (!songId) return;
    try {
      const res = await fetch(`/api/v1/lyrics/${songId}`);
      const data = await res.json();

      setLyrics(data.lyrics);
    } catch (error) {
      setLyrics("Lyrics not available.");
    }
  };

  // const handleHover = () => {
  //   if (songId) {
  //     fetchLyrcis();
  //   }
  //   setShowLyrics(true);
  // }
  // onMouseEnter={handleHover} 
   // onMouseLeave={() => setShowLyrics(false)}
   // {showLyrics && songId && (
  //  <div>
  //  {lyrics}
  //</div>
// )}
  console.log("Cover URL:", cover);

  return <div className="pb-5">
    <img src={cover} alt="cover art" className="block w-100 rounded-lg"></img>
  </div>
}
