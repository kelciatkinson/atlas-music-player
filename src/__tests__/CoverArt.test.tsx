import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import CoverArt from "../components/CoverArt";
import React from "react";

describe("CoverArt Snapshot Tests", () => {
  test("renders with valid cover and songId", () => {
    const { container } = render(
      <CoverArt cover="https://example.com/cover.jpg" songId="123" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with cover but no songId", () => {
    const { container } = render(
      <CoverArt cover="https://example.com/cover.jpg" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with empty cover string", () => {
    const { container } = render(
      <CoverArt cover="" songId="456" />
    );
    expect(container).toMatchSnapshot();
  });

});
