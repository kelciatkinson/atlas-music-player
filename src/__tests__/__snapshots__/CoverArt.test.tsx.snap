// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Snapshot Tests > renders with cover but no songId 1`] = `
<div>
  <div
    class="pb-5"
  >
    <img
      alt="cover art"
      class="block w-100 rounded-lg"
      src="https://example.com/cover.jpg"
    />
  </div>
</div>
`;

exports[`CoverArt Snapshot Tests > renders with empty cover string 1`] = `
<div>
  <div
    class="pb-5"
  >
    <img
      alt="cover art"
      class="block w-100 rounded-lg"
    />
  </div>
</div>
`;

exports[`CoverArt Snapshot Tests > renders with valid cover and songId 1`] = `
<div>
  <div
    class="pb-5"
  >
    <img
      alt="cover art"
      class="block w-100 rounded-lg"
      src="https://example.com/cover.jpg"
    />
  </div>
</div>
`;
